docs/guide.adoc

```diff
@@ -63,8 +63,9 @@ In this example we see two functions. First, we define a new function and store
 it in the constant variable `greet`. We also use the print-function in the
 function body and print a string to the terminal.
 
-The function block starts with `do` and ends with `end`. The `do` can optionally
-be omitted if the function has a return type. See the following examples.
+The function block starts with `do` and ends with `end`. The `do` can be
+replaced with a `->` if the function returns something. See the following
+examples.
```

Comment:
```suggestion
The function block starts with `do` or `->`, and ends with `end`. `do` indicates no return value.
An `->` means the function returns something. See the following
examples.
```
It read a bit wierd - like arrows were special cases.

Reply:
```
The function block starts with `do` or `->` and ends with `end`. `do` indicates
no return value while `->` means the function returns something. See the
following examples.
```